### DISCUSSION TITLE
[Bug] WezTerm window turns blank after OS sleep

### DISCUSSION THREAD

**Discussion by xaviergmail** [wezterm/wezterm#5329] [Q&A] [answered]
### What Operating System(s) are you running on?

macOS

### WezTerm version

20240128-202157-1e552d76

### Ask your question!

Hi, I'm experiencing a bug but I don't have enough info to create a GH issue on the matter. I'm looking for help gathering enough info to create the bug report.

## What I'm experiencing
I have my macbook docked 24/7. It is set to automatically sleep after some time. I tend to leave my terminal open over the weekend and every Monday, I come back to WezTerm being blank.

I see the titlebar with the running process, I'm able to create new tabs and interact with the underlying shell (running vim changes the title to `vim`). Opening the debug "overlay" also works, but the window remains blank.

This has been quite frustrating because I've had to restart my terminal, but I started using tmux again for things that matter so I at least don't lose things I need when this happens.

This morning, I opened a new WezTerm window and the original window started working again shortly after.

I have a very minimal configuration:
```lua
print("hi!")

-- Pull in the wezterm API
local wezterm = require "wezterm"
local assets = wezterm.config_dir .. "/assets"

-- This table will hold the configuration.
local config = {}

-- In newer versions of wezterm, use the config_builder which will
-- help provide clearer error messages
if wezterm.config_builder then
  config = wezterm.config_builder()
end

-- This is where you actually apply your config choices

-- For example, changing the color scheme:
-- config.color_scheme = 'AdventureTime'
config.color_scheme = 'Tomorrow Night Eighties'

config.keys = {
  {
    key = 'r',
    mods = 'CMD|SHIFT',
    action = wezterm.action.ReloadConfiguration,
  },
}

config.background = {
  {
    source = {
      Gradient = {
        orientation = "Horizontal",
        colors = {
          "#00000C",
          "#000026",
          "#00000C",
        },
        interpolation = "CatmullRom",
        blend = "Rgb",
        noise = 0,
      },
    },
    width = "100%",
    height = "100%",
    opacity = 0.85,
  },
  {
    source = {
      File = { path = assets .. "/blob_blue.gif", speed = 0.3 },
    },
    repeat_x = "Mirror",
    -- width = "100%",
    height = "100%",
    opacity = 0.10,
    hsb = {
      hue = 0.9,
      saturation = 0.9,
      brightness = 0.8,
    },
  },
}

-- and finally, return the configuration to wezterm
return config
```

I see the following error messages when checking logs in `$XDG_RUNTIME`:
```
08:52:43.531  ERROR  wezterm_gui::termwindow::render::paint > paint_pass failed: render_backgrounds: frames.current_frame.lease.get_data: Io error in BlobLease: No such file or directory (os error 2): No such file or directory (os error 2)
08:52:44.020  ERROR  wezterm_gui::termwindow::render::paint > paint_pass failed: render_backgrounds: frames.current_frame.lease.get_data: Io error in BlobLease: No such file or directory (os error 2): No such file or directory (os error 2)
08:52:44.341  ERROR  wezterm_gui::termwindow::render::paint > paint_pass failed: render_backgrounds: frames.current_frame.lease.get_data: Io error in BlobLease: No such file or directory (os error 2): No such file or directory (os error 2)
08:52:44.621  ERROR  wezterm_gui::termwindow::render::paint > paint_pass failed: render_backgrounds: frames.current_frame.lease.get_data: Io error in BlobLease: No such file or directory (os error 2): No such file or directory (os error 2)
08:52:46.131  ERROR  wezterm_gui::termwindow::render::paint > paint_pass failed: render_backgrounds: frames.current_frame.lease.get_data: Io error in BlobLease: No such file or directory (os error 2): No such file or directory (os error 2)
08:52:46.213  ERROR  wezterm_gui::termwindow::render::paint > paint_pass failed: render_backgrounds: frames.current_frame.lease.get_data: Io error in BlobLease: No such file or directory (os error 2): No such file or directory (os error 2)
```

The only thing I can think of is the gif background. My issue is, that file never moves.

I have a lot of questions... What could be causing this? Is macOS unmounting the filesystem before it goes to sleep while wezterm is still rendering? And is it reading the file from disk every frame..? Any ideas as to what I can do to gather more information?

As always, any help is appreciated

**Comment by wez** [comment, marked as answer]
Perhaps your system is cleaning out the blob lease cache area from the temp dir?
As a test, you could try exporting `TMPDIR` to some non-temp dir location and see if the problem goes away.
If so, please file a bug report for this!
